contracts/utils/Votes.sol

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "./Context.sol";
+import "./Counters.sol";
+import "./Checkpoints.sol";
+import "./cryptography/draft-EIP712.sol";
+
+/**
+ * @dev Voting operations.
+ */
+abstract contract Votes is Context, EIP712 {
+    using Checkpoints for Checkpoints.History;
+    using Counters for Counters.Counter;
+
+    bytes32 private constant _DELEGATION_TYPEHASH =
+        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
+    mapping(address => address) _delegation;
+    mapping(address => Checkpoints.History) _userCheckpoints;
+    mapping(address => Counters.Counter) private _nonces;
+    Checkpoints.History _totalCheckpoints;
+
+    /**
+     * @dev Emitted when an account changes their delegate.
+     */
+    event DelegateChanged(address indexed delegator, address indexed fromDelegate, address indexed toDelegate);
+
+    /**
+     * @dev Emitted when a token transfer or delegate change results in changes to an account's voting power.
+     */
+    event DelegateVotesChanged(address indexed delegate, uint256 previousBalance, uint256 newBalance);
+
+    /**
+     * @dev Returns total amount of votes for account.
+     */
+    function getVotes(address account) public view returns (uint256) {
+        return _userCheckpoints[account].latest();
+    }
+
+    /**
+     * @dev Returns total amount of votes at given position.
+     */
+    function _getPastVotes(address account, uint256 timestamp) internal view returns (uint256) {
```

Comment:
Why is there a combination of public (`getVotes`) and internal (`_getPastVotes`) functions? I think they should all be public functions.

Reply:
`getVotes` returns the amount of votes for an account, and `_getPastVotes` returns the total amount on a blockNumber.